.github/workflows/pre-commit-ci.yml

```diff
@@ -0,0 +1,19 @@
+name: pre-commit-ci
+on:
+  pull_request:
+    branches: [master]
+jobs:
+  pre-commit-ci:
+    name: Run pre-commit checks on changed files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: git fetch --depth=1 origin +${{github.base_ref}}
```

Comment:
Really appreciated the effort towards gradual testing! Consider adding a comment above this line to provide more context and help recall the purpose.